tests/network/test_topology.py

```diff
@@ -10,32 +10,163 @@
 from edisgo import EDisGo
 from edisgo.network.grids import LVGrid
 
+from edisgo.network.components import Switch
 
 class TestTopology:
     """
     Tests Topology class, except methods that require an edisgo object.
 
     """
+
     @classmethod
     def setup_class(self):
         self.topology = Topology()
         ding0_import.import_ding0_grid(pytest.ding0_test_network_path, self)
 
     def test_rings(self):
-        """ToDo: Test rings getter."""
-        pass
+        """Test rings getter."""
+
+        # getting rings
+        ring = self.topology.rings
+        # sorting ring elements, as they have a different order on each pass
+        ring[0].sort()
+        ring[1].sort()
+        ring.sort()
+
+        rings_ding0_test_network_1 = [['BusBar_MVGrid_1_LVGrid_1_MV',
+                                       'BusBar_MVGrid_1_LVGrid_5_MV',
+                                       'BusBar_MVGrid_1_LVGrid_6_MV',
+                                       'BusBar_MVGrid_1_LVGrid_9_MV',
+                                       'Bus_BranchTee_MVGrid_1_1',
+                                       'Bus_BranchTee_MVGrid_1_2',
+                                       'Bus_BranchTee_MVGrid_1_3',
+                                       'Bus_BranchTee_MVGrid_1_4',
+                                       'Bus_MVStation_1'],
+                                      ['BusBar_MVGrid_1_LVGrid_4_MV',
+                                       'BusBar_MVGrid_1_LVGrid_8_MV',
+                                       'Bus_BranchTee_MVGrid_1_10',
+                                       'Bus_BranchTee_MVGrid_1_11',
+                                       'Bus_BranchTee_MVGrid_1_5',
+                                       'Bus_BranchTee_MVGrid_1_6',
+                                       'Bus_BranchTee_MVGrid_1_7',
+                                       'Bus_BranchTee_MVGrid_1_8',
+                                       'Bus_BranchTee_MVGrid_1_9',
+                                       'Bus_MVStation_1']]
+        # test if rings have expected elements
+        assert ring == rings_ding0_test_network_1
 
     def test_get_connected_lines_from_bus(self):
-        """ToDo: Test get_connected_lines_from_bus method."""
-        pass
+        """Test get_connected_lines_from_bus method."""
+
+        # get connected lines
+        connected_lines = self.topology.get_connected_lines_from_bus(
+            "Bus_BranchTee_MVGrid_1_8"
+        )
+        # test if the expected lines are connected
+        assert "Line_10019" in connected_lines.index
+        assert "Line_10020" in connected_lines.index
+        assert "Line_10021" in connected_lines.index
+        # test if the selected Bus is connected to the found lines
+        assert "Bus_BranchTee_MVGrid_1_8" in connected_lines["bus0"].values
```

Comment:
If you want to test if the bus is found in the connected lines' attributes, you should for each line test if the bus is in either bus0 or bus1.